Dockerfile

```diff
@@ -16,8 +16,13 @@ COPY controllers/ controllers/
 COPY internal/ internal/
 COPY cmd/ cmd/
 
+ARG VERSION_PKG
+ARG COMMIT_SHA
+ARG OPERATOR_VERSION
+ARG VERSION_DATE
+
 # Build
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o manager main.go
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -ldflags="-X ${VERSION_PKG}.buildDate=${VERSION_DATE} -X ${VERSION_PKG}.version=${OPERATOR_VERSION} -X ${VERSION_PKG}.commitSha=${COMMIT_SHA}" -a -o manager main.go
```

Comment:
Would it not make sense to call `make build` here instead?

Reply:
I'm following the same approach followed in OpenTelemetry Operator and Jaeger Operator. Should we change the 3 of them?

Reply:
I assume we should start here. But sure, why not.

Reply:
I asked to create the same issue for OTEL and Jaeger Operators. TBH, I was thinking about this change but wanted to follow the same approach as in the other projects because I thought you would prefer consistency.